Lib/typing.py

```diff
@@ -1,20 +1,21 @@
 """
-The typing module: Support for gradual typing as defined by PEP 484.
+The typing module: Support for gradual typing as defined by PEP 484 and subsequent PEPs.
 
 At large scale, the structure of the module is following:
 * Imports and exports, all public names should be explicitly added to __all__.
 * Internal helper functions: these should never be used in code outside this module.
 * _SpecialForm and its instances (special forms):
-  Any, NoReturn, Never, ClassVar, Union, Optional, Concatenate, Unpack
-* Classes whose instances can be type arguments in addition to types:
-  ForwardRef, TypeVar and ParamSpec
+  NoReturn, Never, ClassVar, Self, Concatenate, Unpack, and others.
+* Classes whose instances can be type arguments:
+  TypeVar, ParamSpec, TypeVarTuple.
```

Comment:
- `Any` is now its own class, rather than being an instance of `_SpecialForm`
- Numerous other `_SpecialForm`s have now been added. Rather than attempting to list them all (which would just go out of date again), I've just added "and others" at the end of an abbreviated list.
- Instances of `ForwardRef` aren't actually accepted by type checkers as type annotations, as far as I know, so that was somewhat confusing here.